pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -341,6 +352,25 @@ func (r *controlPlaneInstanceReconciler) Reconcile(ctx context.Context) (result
 	return
 }
 
+// finds the first SMCP in the given namespace by creationTimestamp
+func getEarliestSMCPInNamespace(ctx context.Context, cl client.Client, ns string) (*v2.ServiceMeshControlPlane, error) {
+	smcpList := &v2.ServiceMeshControlPlaneList{}
+	if err := cl.List(ctx, smcpList, client.InNamespace(ns)); err != nil {
+		return nil, err
+	}
+
+	var earliest *v2.ServiceMeshControlPlane
+	for _, smcp := range smcpList.Items {
+		if earliest == nil ||
+			smcp.CreationTimestamp.Before(&earliest.CreationTimestamp) ||
+			(smcp.CreationTimestamp == earliest.CreationTimestamp && smcp.Name < earliest.Name) {
```

Comment:
What's the use case here? Is this even possible? (to have two equal timestamps for the same resource)
Why the name is important here?

Reply:
> Why the name is important here?

I think that when timestamps are equal, you can't determine which one should be reconciled, so name is an additional parameter for selecting the right object.

Reply:
but name should not matter. alphabetical order doesn't mean anything. Or does it?

Reply:
ok, why not use random() instead of going alphabetical. What if 1st name is `b` and second name is `a`?

Reply:
I mean, random() is a joke... I'm just trying to understand why we have this comparison at all. If timestamps are equal, pick any of those.

Reply:
We can't rely on resourceVersion for this, right?

Reply:
Which SMCP gets reconciled must be deterministic. If we were to choose which SMCP to reconcile at random, the controller could reconcile both SMCPs or neither. 

Think about what happens when we have two SMCPs with the same timestamp. 
1. The controller is asked to reconcile SMCP1. It chooses the earliest SMCP at random. Let's say it chooses SMCP2 this time, so it skips reconciliation of SMCP1 and adds the error to its status.
2. The controller is then asked to reconcile SMCP2. This time, it chooses SMCP as the earliest, so it skips reconciliation of SMCP2 and adds the error to it. 

Neither SMCP gets reconciled. In another scenario, the controller could reconcile both SMCPs. 

That's why we need to sort by name (or some other field), so that the same SMCP is always chosen as the one to reconcile.

Reply:
`resourceVersion` is supposed to be opaque, meaning that we shouldn't rely on its value. All we're allowed to do is check if two resourceVersions are equal. 

It's possible that the format of `resourceVersion` will change in the future. It can already be different if you use a datastore other than etcd.